examples/meeting-transcription/meeting_transcript_example/steps/meeting-transcription-api.step.ts

```diff
@@ -0,0 +1,146 @@
+/**
+ * Meeting Transcription API Step
+ * 
+ * This step demonstrates how to create an API endpoint in Motia that:
+ * 1. Accepts HTTP requests for meeting transcription
+ * 2. Validates input using Zod schemas
+ * 3. Emits events for asynchronous processing
+ * 4. Creates real-time streaming updates
+ * 5. Returns structured responses
+ */
+
+import { ApiRouteConfig, Handlers } from 'motia'
+import { z } from 'zod'
+
+export const config: ApiRouteConfig = {
+  type: 'api',
+  name: 'MeetingTranscriptionAPI',
+  description: 'API endpoint for meeting transcription requests with real-time streaming',
+
+  method: 'POST',
+  path: '/transcribe-meeting',
+
+  /**
+   * This API Step emits events to topic `meeting-transcription`
+   * The event will be processed asynchronously by the MeetingTranscriptionProcessor
+   */
+  emits: ['meeting-transcription'],
+
+  /** 
+   * Input validation using Zod schemas
+   * This ensures type safety and provides automatic documentation
+   */
+  bodySchema: z.object({ 
+    filename: z.string().describe('Name of the audio file to transcribe'),
+    audioData: z.string().optional().describe('Base64 encoded audio data (optional)'),
```

Comment:
note: this works but is not ideal to receive file uploads as base64 for performance reasons as it loads the memory and big file uploads like 100MB consume just too much memory to handle scalability

ideal scenario would be sending a Pre-signed URL for the frontend to upload to Supabase Storage or Amazon S3 directly